java/DynamoDBStreamSource/src/main/java/com/amazonaws/services/msf/StreamingJob.java

```diff
@@ -0,0 +1,75 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Properties;
+
+public class StreamingJob {
+    private static final Logger LOG = LoggerFactory.getLogger(StreamingJob.class);
+
+    // Name of the local JSON resource with the application properties in the same format as they are received from the Amazon Managed Service for Apache Flink runtime
+    private static final String LOCAL_APPLICATION_PROPERTIES_RESOURCE = "flink-application-properties-dev.json";
+
+    private static boolean isLocal(StreamExecutionEnvironment env) {
+        return env instanceof LocalStreamEnvironment;
+    }
+
+    /**
+     * Load application properties from Amazon Managed Service for Apache Flink runtime or from a local resource, when the environment is local
+     */
+    private static Map<String, Properties> loadApplicationProperties(StreamExecutionEnvironment env) throws IOException {
+        if (isLocal(env)) {
+            LOG.info("Loading application properties from '{}'", LOCAL_APPLICATION_PROPERTIES_RESOURCE);
+            return KinesisAnalyticsRuntime.getApplicationProperties(
+                    StreamingJob.class.getClassLoader()
+                            .getResource(LOCAL_APPLICATION_PROPERTIES_RESOURCE).getPath());
+        } else {
+            LOG.info("Loading application properties from Amazon Managed Service for Apache Flink");
+            return KinesisAnalyticsRuntime.getApplicationProperties();
+        }
+    }
+
+    private static DynamoDbStreamsSource<ChangeEvent> getDynamoDbStreamsSource(Properties sourceProperties) {
+        Configuration sourceConfig = Configuration.fromMap(Maps.fromProperties(sourceProperties));
+        String streamArn = sourceProperties.getProperty("stream.arn");
+
+        return DynamoDbStreamsSource.<ChangeEvent>builder()
+                .setStreamArn(streamArn)
+                .setSourceConfig(sourceConfig)
+                // User must implement their own deserialization schema to translate change data capture events into custom data types
+                .setDeserializationSchema(new ChangeEventDeserializationSchema())
+                .build();
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        // set up the streaming execution environment
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        final Map<String, Properties> applicationProperties = loadApplicationProperties(env);
+        LOG.warn("Application properties: {}", applicationProperties);
+
+        DynamoDbStreamsSource<ChangeEvent> source = getDynamoDbStreamsSource(applicationProperties.get("InputStream0"));
+        env.fromSource(source, WatermarkStrategy.forMonotonousTimestamps(), "DynamoDb Streams Source")
+                .returns(TypeInformation.of(ChangeEvent.class))
+                .map(changeEvent -> {
```

Comment:
We need to put a comment explaining this is for the sake of this example but you should never log every record in a production application, because it may seriously impact performance (etc etc). Remember everything that appears in these example is considered best practice.
Also, make it debug level, just in case :)

Reply:
Ok thats a good shout. I will remove this and change to KDS output.